source/compiler/qsc/src/packages.rs

```diff
@@ -75,19 +76,70 @@ fn convert_circuit_sources(
     processed_sources
 }
 
+#[must_use]
+pub fn get_target_profile_from_entry_point(
+    package_graph_sources: &PackageGraphSources,
+    dependency_errors: &mut Vec<WithSource<ErrorKind>>,
+) -> Option<Profile> {
+    // Convert circuit files in user code to generated Q# before entry profile check
+    let mut sources = package_graph_sources.root.sources.clone();
+    sources = convert_circuit_sources(sources, dependency_errors);
```

Comment:
Since you already call `convert_circuit_sources` in `prepare_package_store`, could you perhaps perform this check on those converted sources and raise the error there?

Reply:
I couldn't quite get the logic to a place where the error could be thrown where you suggested. The best I'm able to do is not do the circuit conversion when looking for target profile. Because circuit files won't have such a thing, I filtered them out of that search.